frontend/src/components/custom-tools/tools-main/ToolsMainActionBtns.jsx

```diff
@@ -52,6 +65,13 @@ function ToolsMainActionBtns() {
   return (
     <Space>
       {singlePassExtractMode && RunSinglePassBtn && <RunSinglePassBtn />}
+      {singlePassExtractMode && ChallengeModal && (
+        <ChallengeModal
+          challengeData={tokenUsage?.[`${tokenUsageId}__challenge_data`]}
+          context={tokenUsage?.[`${tokenUsageId}__context`]}
+          tokenUsage={tokenUsage?.[`${tokenUsageId}`]}
```

Comment:
@Deepak-Kesavan We don't need to use template literal for this usecase. It can be directly added like below,

```suggestion
          tokenUsage={tokenUsage?.[tokenUsageId]}
```